### ISSUE TITLE
Float32 targets support 

### ISSUE THREAD

**Issue by dkaluza** [CDonnerer/xgboost-distribution#97]
Hi,

Thanks for the great library.

I have tried to use the model with  float32  array targets and have encountered the following error:

TypeError: Object of type float32 is not JSON serializable

Minimal reproduction example:
```
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split

from xgboost_distribution import XGBDistribution


data = fetch_california_housing()
X, y = data.data, data.target
X_train, X_test, y_train, y_test = train_test_split(X, y)

model = XGBDistribution(
    distribution="normal",
    n_estimators=500,
    early_stopping_rounds=10
)
model.fit(X_train, y_train.astype('float32'), eval_set=[(X_test, y_test)])
```

My relevant library stack:
xgboost                   2.0.2
numpy                     1.26.2
xgboost-distribution      0.2.8

Looks like issue is just serialization related as it is thrown after booster training.

I can try to make an appropriate hotfix myself and make a MR, but it will probably require some investigation from my side to make it properly. Let me know if I should proceed.

**Comment by CDonnerer** [comment]
Hi,
Thanks for raising.

I'll put in a fix, however, it's worth noting that xgboost internally uses 32 bit floats (as does xgboost-distribution), so it will already get converted.

**Comment by CDonnerer** [comment]
Fixed in `xgboost-distribution>=0.2.9`
